frontend/context/SharedProvider/SharedProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import { createContext, PropsWithChildren, useCallback, useRef } from 'react';
+
+import { Optional } from '@/types/Util';
+
+import { useMainOlasBalance } from './useMainOlasBalance';
+
+export const SharedContext = createContext<{
+  // main olas balance
+  isMainOlasBalanceLoading: boolean;
+  mainOlasBalance: Optional<number>;
+  hasMainOlasBalanceAnimatedOnLoad: boolean;
+  setMainOlasBalanceAnimated: (value: boolean) => void;
+
+  // others
+}>({
+  isMainOlasBalanceLoading: true,
+  mainOlasBalance: undefined,
+  hasMainOlasBalanceAnimatedOnLoad: false,
+  setMainOlasBalanceAnimated: () => {},
+});
+
+/**
+ * Shared provider to provide shared context to all components in the app.
+ * @example
+ * - Track the main OLAS balance animation state & mount state.
+ */
+export const SharedProvider = ({ children }: PropsWithChildren) => {
```

Comment:
shouldn't we call it SharedOlasBalanceProvider? or if it's specifically used for animation, call it like that? otherwise I think we create another layer having already too many